src/agent/src/mount.rs

```diff
@@ -309,8 +309,33 @@ pub async fn update_ephemeral_mounts(
 async fn overlayfs_storage_handler(
     logger: &Logger,
     storage: &Storage,
+    cid: Option<&str>,
     _sandbox: Arc<Mutex<Sandbox>>,
 ) -> Result<String> {
+    if storage
+        .options
+        .iter()
+        .any(|e| e == "io.katacontainers.fs-opt.overlay-rw")
+    {
+        let cid = cid.ok_or_else(|| anyhow!("No container id in rw overlay"))?;
+        let cpath = Path::new(crate::rpc::CONTAINER_BASE).join(cid);
+        let work = cpath.join("work");
+        let upper = cpath.join("upper");
+
+        fs::create_dir_all(&work)?;
```

Comment:
Maybe add a `.context("Creating work directory")` and similarly for next line, since there are multiple places that can fail in this function

Reply:
Good idea. Will do.